src/side_chain.cpp

```diff
@@ -598,15 +598,18 @@ void SideChain::print_status()
 	uint32_t total_blocks_in_window = 0;
 	uint32_t total_uncles_in_window = 0;
 
-	uint32_t our_blocks_in_window = 0;
-	uint32_t our_uncles_in_window = 0;
+	// each dot corresponds to m_chainWindowSize / 30 shares, with current values, 2160 / 30 = 72
+	std::vector<uint32_t> our_blocks_in_window(30, 0);
+	std::vector<uint32_t> our_uncles_in_window(30, 0);
 
 	while (cur) {
 		blocks_in_window.emplace_back(cur->m_sidechainId);
 		++total_blocks_in_window;
 
 		if (cur->m_minerWallet == m_pool->params().m_wallet) {
-			++our_blocks_in_window;
+			// this produces an integer division with quotient rounded up, avoids non-whole divisions from overflowing on total_blocks_in_window
+			const size_t window_index = (total_blocks_in_window - 1) / ((m_chainWindowSize + our_blocks_in_window.size() - 1) / our_blocks_in_window.size());
+			our_blocks_in_window[std::min(window_index, our_uncles_in_window.size() - 1)]++; // clamp window_index, even if total_blocks_in_window is not larger than m_chainWindowSize
```

Comment:
our_uncles_in_window.size() - copy/paste error?

Reply:
Correct, fixed a math error and copy&pasted, will force-push a change